### PR TITLE
 Add support for BERDY_FLOATING_BASE in BerdySparseMAPSolver 

### PR DESCRIPTION
**Pull request by traversaro**
Also added a consistency test between `BerdySparseMAPSolver` and `ExtWrenchesAndJointTorquesEstimator` .

### CODE REVIEW

**Comment by traversaro** [comment]
The only test failing is the octave one. Let me know when you think that this is ready to be merged, so I can regenerate and commit the bindings.

File: src/core/include/iDynTree/Core/TestUtils.h [hunk 1]
```diff
@@ -311,6 +311,46 @@ namespace iDynTree
...
+            {
+                checkCorrect = false;
+                correctElements[i] = false;
+            }
+        }
+
+        if( !checkCorrect )
+        {
+            std::cerr << file << ":" << line << " : assertVectorAreEqualWithRelativeTol failure: " << std::endl;
+            printVector("vec1",vec1);
```
Comment (francesco-romano) [inline src/core/include/iDynTree/Core/TestUtils.h]: Comment for a later PR: I think we could add a newline between the various prints, so we understand when a vector ends and the other starts

Comment (traversaro) [reply src/core/include/iDynTree/Core/TestUtils.h]: Yes, I am also always bothered by that.

**Comment by traversaro** [comment]
This branch is finally fully working. I rebased away the bindings to have a branch easy to review. @francesco-romano @diegoferigo if you want you can check the latest 5/6 commits, but it would be great to merge this as soon as possible to avoid the branch balkanization.

**Comment by traversaro** [comment]
(Bindings need to be regenerated before merging).

File: src/estimation/src/BerdyHelper.cpp [hunk 2]
```diff
@@ -542,6 +543,11 @@ IndexRange BerdyHelper::getRangeSensorVariable(const SensorType type, const unsi
         sensorOffset += 3*m_sensors.getNrOfSensors(THREE_AXIS_ANGULAR_ACCELEROMETER);
     }
 
+    if( type > THREE_AXIS_FORCE_TORQUE_CONTACT )
```
Comment (francesco-romano) [inline src/estimation/src/BerdyHelper.cpp]: Sorry.. I probably saw this method a lot of time, but only now I am wondering.. why `type > XXX` ?

File: src/estimation/src/BerdyHelper.cpp [hunk 3]
```diff
@@ -1270,9 +1300,11 @@ bool BerdyHelper::computeBerdySensorMatrices(SparseMatrix<iDynTree::ColumnMajor>
                 IndexRange sensorRange = this->getRangeLinkSensorVariable(NET_EXT_WRENCH_SENSOR,idx);
                 IndexRange netExtWrenchRange = this->getRangeLinkVariable(NET_EXT_WRENCH,idx);
 
-                matrixYElements.addDiagonalMatrix(sensorRange,
-                                                  netExtWrenchRange,
-                                                  1);
+                Transform measurementFrame_X_link = (m_link_H_externalWrenchMeasurementFrame[idx].inverse());
```
Comment (francesco-romano) [inline src/estimation/src/BerdyHelper.cpp]: parenthesis should not be needed (also line 1284)

File: src/sensors/src/ThreeAxisForceTorqueContactSensor.cpp [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+    Wrench totalWrench;
+    totalWrench.zero();
+
+    for (size_t i=0; i < loadCellMeasurements.size(); i++)
+    {
+        // We assume that the total forcetorque is given by the sum of pure forces on the z axis measured by the load cells
+        Wrench loadCellWrench = Wrench(LinearForceVector3(0.0, 0.0, loadCellMeasurements(i)*1.0), AngularForceVector3(0.0, 0.0, 0.0));
+        Transform sensor_H_loadCell(Rotation::Identity(), pimpl->m_loadCellLocations[i]);
+        totalWrench = totalWrench + sensor_H_loadCell*loadCellWrench;
```
Comment (francesco-romano) [inline src/sensors/src/ThreeAxisForceTorqueContactSensor.cpp]: nitpick: I usually find more compact (but also readable) the `+=` operator

File: src/sensors/src/ThreeAxisForceTorqueContactSensor.cpp [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+Position ThreeAxisForceTorqueContactSensor::computeCenterOfPressureFromLoadCellMeasurements(VectorDynSize& loadCellMeasurements)
+{
+    if (loadCellMeasurements.size() != pimpl->m_loadCellLocations.size())
+    {
+        reportError("ThreeAxisForceTorqueContactSensor","computeThreeAxisForceTorqueFromLoadCellMeasurements", "loadCellMeasurements has the wrong size");
+        Position zeroVec; zeroVec.zero();
+        return zeroVec;
+    }
+
+    // We assume that the location of the load cells is on the XY plane. In this case, the cop is just the average of the locations weighted on the measurements
```
Comment (francesco-romano) [inline src/sensors/src/ThreeAxisForceTorqueContactSensor.cpp]: Just a question: this should also match to the result computed by `computeThreeAxisForceTorqueFromLoadCellMeasurements` and properly manipulated, right? (like torque / f_z)
Can it be used as a consistency test?

File: src/sensors/src/ThreeAxisForceTorqueContactSensor.cpp (hunk 4 above)
Comment (traversaro) [reply src/sensors/src/ThreeAxisForceTorqueContactSensor.cpp]: There is no `Wrench::operator+=` at the moment. : )

File: src/sensors/src/ThreeAxisForceTorqueContactSensor.cpp (hunk 5 above)
Comment (traversaro) [reply src/sensors/src/ThreeAxisForceTorqueContactSensor.cpp]: Yes, indeed.

File: src/estimation/include/iDynTree/Estimation/BerdyHelper.h [hunk 6]
```diff
@@ -699,6 +723,22 @@ class BerdyHelper
...
+     * net wrench is known (for example when it is known that the external net wrench is a pure force on a point different from the
+     * link frame.
+     *
+     * \note This will only change the frame in which the measurent equation of the net external wrench is expressed,
+     *       not how the Newton-Euler equation of the link are expressed or how the net ext wrenches are serialized in
+     *       the LinkNetExternalWrenches class.
+     */
+    ///@{
+    bool setNetExternalWrenchMeasurementFrame(const LinkIndex lnkIndex, const Transform& link_H_contact);
+    bool getNetExternalWrenchMeasurementFrame(const LinkIndex lnkIndex, Transform& link_H_contact);
```
[the remark this answers is not in the document]
Comment (traversaro) [reply src/estimation/include/iDynTree/Estimation/BerdyHelper.h]: Good point.

File: src/estimation/src/BerdyHelper.cpp (hunk 2 above)
Comment (traversaro) [reply src/estimation/src/BerdyHelper.cpp]: Basically is quite a crude way to express the logic "if the sensor comes after all the sensors of type XXX in serialization, then you have to use the proper offset when computing the sensor range in the sensor serialization.

File: src/sensors/src/ThreeAxisForceTorqueContactSensor.cpp (hunk 4 above)
Comment (francesco-romano) [reply src/sensors/src/ThreeAxisForceTorqueContactSensor.cpp]: I did't realise it was on the Wrench object. Ok :)

**Comment by traversaro** [comment]
@francesco-romano The latest two commits should have addressed your comments. Let me know if they are ok so I can do the final rebase+bindings generation.

**Comment by francesco-romano** [comment]
👍  Go with the merge

**Pull request merged**